### PR TITLE
Enable CI and add job for running cargo risczero test

### PR DESCRIPTION
**Pull request by nategraf**
This PR enables CI to run by changing the source of the Rust toolchain action to come from the `risc0` repository. It also adds a job which uses the WIP `cargo risczero test` command to run all unit tests inside of the RISC Zero zkVM executor.

Depends on https://github.com/risc0/risc0/pull/646

### CODE REVIEW

**Review by flaub — COMMENTED** [review]
LGTM

File: .github/workflows/crypto-bigint.yml [hunk 1]
```diff
@@ -142,7 +147,21 @@ jobs:
...
+  risczero-test:
+    name: Risc Zero Test
+    runs-on: [self-hosted, prod, Linux, cpu]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: risc0/risc0/.github/actions/rustup@main
+      # install the risc0 toolchain using the latest release of cargo-risczero
+      - run: cargo install cargo-risczero
+      - run: cargo risczero install
+      - run: git clone https://github.com/risc0/risc0.git
```
Comment (flaub) [inline .github/workflows/crypto-bigint.yml]: We should use the checkout action instead of git directly

**Comment by flaub** [comment]
I'm thinking we should drop the checks that involve tests with other targets, since this can eat up lots of CI resources.

**Comment by nategraf** [comment]
I used the `cargo risczero test` feature built with `risc0` at `main` (as of Nov 13) and hit a new issue. Execution of the tests works as intended, but ends with `Error: address 0x00000000 is an invalid guest address`. `cargo risczero` then exits with status code `1`.

File: .github/workflows/crypto-bigint.yml [hunk 2]
```diff
@@ -27,7 +32,7 @@ jobs:
           - wasm32-unknown-unknown
     steps:
       - uses: actions/checkout@v3
-      - uses: dtolnay/rust-toolchain@master
+      - uses: risc0/risc0/.github/actions/rustup@main
```
Comment (flaub) [inline .github/workflows/crypto-bigint.yml:35]: I think we should avoid pointing to `main`

File: .github/workflows/crypto-bigint.yml [hunk 3]
```diff
@@ -142,7 +147,24 @@ jobs:
...
+        with:
+          repository: risc0/risc0
+          path: risc0
+      - uses: risc0/risc0/.github/actions/rustup@main
+      # install the risc0 toolchain using the latest release of cargo-risczero
+      - run: cargo install cargo-risczero
+      - run: cargo risczero install
+      # rebuild with the latest cargo-risczero with the experimental feature
+      - run: cargo install cargo-risczero --path risc0/risc0/cargo-risczero --features experimental
+      - run: cargo risczero test -- --release --all-features
```
Comment (flaub) [inline .github/workflows/crypto-bigint.yml:170]:
```suggestion
      - run: cargo install cargo-risczero --path risc0/risc0/cargo-risczero --features experimental
      - run: cargo risczero install
      - run: cargo risczero test -- --release --all-features
```

Building twice will be pretty slow

File: .github/workflows/crypto-bigint.yml [hunk 4]
```diff
@@ -142,7 +147,24 @@ jobs:
...
+  risczero-test:
+    name: Risc Zero Test
+    runs-on: [self-hosted, prod, Linux, cpu]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+        with:
+          repository: risc0/risc0
+          path: risc0
+      - uses: risc0/risc0/.github/actions/rustup@main
```
Comment (flaub) [inline .github/workflows/crypto-bigint.yml:164]: Do we want to build against `main`?
